dhis-2/dhis-services/dhis-service-dxf2/src/main/java/org/hisp/dhis/dxf2/metadata/objectbundle/validation/attribute/DefaultAttributeValidator.java

```diff
@@ -92,19 +70,55 @@
    */
   @Override
   public void validate(ValueType valueType, String value, Consumer<ErrorReport> addError) {
-    if (StringUtils.isEmpty(value)) {
-      return;
+    if (isEmpty(value)) return;
+    switch (valueType) {
```

Comment:
This is a valid point but addressing it will change behaviour in a potentially incompatible way. The errors are essentially found eagerly when a value is set/updated instead of when a value is converted on usage. We should add the missing cases but it might be better to do this in a separate issue.